lib/adagrams.rb

```diff
@@ -0,0 +1,125 @@
+def draw_letters
+  letter_distribution = {
+    "a" => 9,
+    "b" => 2,
+    "c" => 2,
+    "d" => 4,
+    "e" => 12,
+    "f" => 2,
+    "g" => 3,
+    "h" => 2,
+    "i" => 9,
+    "j" => 1,
+    "k" => 1,
+    "l" => 4,
+    "m" => 2,
+    "n" => 6,
+    "o" => 8,
+    "p" => 2,
+    "q" => 1,
+    "r" => 6,
+    "s" => 4,
+    "t" => 6,
+    "u" => 4,
+    "v" => 2,
+    "w" => 2,
+    "x" => 1,
+    "y" => 2,
+    "z" => 1,
+  }
+  letter_pool = []
+  letter_distribution.each do |str, n|
+    n.times do
+      letter_pool << str.downcase
+    end
+  end
+  hand = letter_pool.shuffle[0..9]
+  return hand
+end
+
+def uses_available_letters?(input, letters_in_hand)
+  if input.length > 10
+    puts "Too many letters! Try again: "
+    input = gets.chomp.downcase
```

Comment:
You shouldn't be handing input from the user here.  The `wave-N-game` should handle that for you.

The reason for this is that it's tricky to test code that asks for user input so it's nice to have that all in one place.